system/util.rs

```diff
@@ -0,0 +1,212 @@
+pub mod transaction_util {
+    use concept::thing::thing_manager::ThingManager;
+    use concept::type_::type_manager::TypeManager;
+    use database::transaction::{DataCommitError, SchemaCommitError, TransactionRead, TransactionSchema, TransactionWrite};
+    use database::Database;
+    use function::function_manager::FunctionManager;
+    use lending_iterator::LendingIterator;
+    use options::TransactionOptions;
+    use query::query_manager::QueryManager;
+    use std::sync::Arc;
+    use storage::durability_client::WALClient;
+    use storage::snapshot::{SchemaSnapshot, WriteSnapshot};
+
+    #[derive(Debug)]
+    pub struct TransactionUtil {
+        database: Arc<Database<WALClient>>,
+    }
+
+    impl TransactionUtil {
+        pub fn new(database: Arc<Database<WALClient>>) -> Self {
+            TransactionUtil { database }
+        }
+
+        pub fn schema_transaction<T>(
```

Comment:
It really looks like `with_schema_tx!` macro from our tests. Might be cleaner to reuse these macros. Another `Util` class (which's instance is called `mgr`) feels heavily redundant, but might fine for the `system` package... 
I'm not going to fight over it considering our timeline though, but some TODOs can look good here.

Reply:
Yeah agreed we could look how to generalise them. The reason I'm not using `with_schema_tx!` is because it's not quite generic and makes some assumptions about some test context.